### PR TITLE
Adding limited CI job to run against staging Terra orchestration API (SCP-5196)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND
Our current integration with Terra is such that SCP only ever points at their production instance.  This was done for a variety of reasons, mostly for the uptime commitments and QA/QC.  However, a few times this has led to a scenario where a breaking API change is released to production, which causes all SCP instances to encounter some level of degraded service, depending on the change/regression.

#### CHANGES
This update adds a "smoke test" that will run a limited CI job (only `fire_cloud_client_test.rb`) against the [staging](https://firecloud-orchestration.dsde-staging.broadinstitute.org) instance of orchestration.  By doing so, we aim to catch regressions _before_ they are released.  Staging is the appropriate instance as it is up consistently and receives updates weekly that are intended to be released to production.  This job will only run on commits to `development` or `master`, but also could be configured to run on a regular cadence (nightly, weekly, etc).  

In addition, this update contains other changes to support this work.  Most notably, `FireCloudClient` now uses keyword parameters instead of positional ones to make instantiating clients simpler.  Additionally, `FireCloudProfile` now can be used to make updates to your Terra profile from the console to aid in testing this work.  The necessary projects/permissions were created outside of this PR directly in the staging Terra instance.

#### MANUAL TESTING
In addition to the manual test instructions below, you can see in the "Checks" portion of this PR that the [CI job](https://github.com/broadinstitute/single_cell_portal_core/actions/runs/5591300236) has run successfully.

1. Pull this branch and enter a Rails console session
2. Register your service account in the staging SAM instance, adding in your email address for `contactEmail`:
```
client = FireCloudClient.new(
  api_root: 'https://firecloud-orchestration.dsde-staging.broadinstitute.org'
)
profile = FireCloudProfile.new(
  contactEmail: "< your email >",                                                                             
  email: client.issuer,
  firstName: "SCP Dev",
  lastName: "Service Account",
  institute: "Broad Institute",
  institutionalProgram: "DSP",
  nonProfitStatus: "true"",
  pi: "N/A",
  programLocationCity: "Cambridge",
  programLocationState: "MA",
  programLocationCountry: "USA",
  title: "N/A"
)
client.set_profile(profile.attributes)
```
3. In a separate terminal, run the smoke test shell script from the project root: 
```
# -d flag will run test outside of Docker
./rails_local_setup.rb && source config/secrets/.source_env.bash
bin/run_orch_smoke_test.sh -d
```

You should see similar output to this (note the logging statement that points to staging orchestration):
```
Run options: --seed 17362

# Running:

SUITE FireCloudClientTest starting
Running smoke test against Terra orchestration API at https://firecloud-orchestration.dsde-staging.broadinstitute.org
fire_cloud_client_test.rb:76 test_firecloud_api_available starting
...
```

### CODE REVIEW

**Review by eweitz — APPROVED** [review]
Code looks good.  Detecting upstream breaks earlier is valuable!  

> This job will only run on commits to development or master, but also could be configured to run on a regular cadence (nightly, weekly, etc).

Future enhancement idea:

I suspect it'd be most efficient and robust to _only_ run on a regular cadence, and not on push to `development` or `master`.  The benefit of this change is protecting study creation from breaking.  The cost is a longer developer wait times for PRs to be green-lit to merge, due to longer CI times, due to more saturated CI build queue.

The [observed impact](https://broadinstitute.slack.com/archives/C02LMB1PQ6R/p1686749263398779?thread_ts=1686685672.683629&cid=C02LMB1PQ6R) of the upstream bug on SCP was quite low.  That chart reports production had 1 affected user, 1 time; but reported release times actually imply 0 users were affected on production.  

Switching this build trigger from push to, say, a regular daily time would (IIUC) make this CI job _non-blocking_ for PR merges, and thus not unnecessarily impede merge flow and perhaps usually detect upstream breaks even earlier.

File: .github/workflows/run-orch-smoketest.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          # add ci- prefix to avoid post-merge build failing due to missing GITHUB_HEAD_REF
+          # and strip any slashes from GITHUB_HEAD_REF to avoid 'invalid reference format' error with tag
+          # this will happen on dependabot PRs
+          SAFE_HEAD_REF=$(echo $GITHUB_HEAD_REF | sed -e 's/\//_/g')
+          DOCKER_IMAGE_TAG="ci-$SAFE_HEAD_REF-$GITHUB_SHA"
+          docker build -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG -f test/Dockerfile-test .
+          bin/load_env_secrets.sh -p secret/kdux/scp/staging/scp_config.json \
+                                  -s secret/kdux/scp/staging/scp_service_account.json \
+                                  -e test -v $DOCKER_IMAGE_TAG -n single-cell-portal-test \
+                                  -c bin/run_orch_smoke_test.sh
```
[the remark this answers is not in the document]
Comment (bistline) [reply .github/workflows/run-orch-smoketest.yml:61]: Ah yes, a composite action looks like the correct path now.  I'll give that a shot.  Thanks!

**Comment by bistline** [comment]
> Switching this build trigger from push to, say, a regular daily time would (IIUC) make this CI job _non-blocking_ for PR merges. That, in turn, would not impede merge flow and perhaps usually detect upstream breaks even earlier.

The way the build is configured now, it would only run _after_ the PR merged, meaning it is non-blocking.  It only runs on commits because I added this branch to the `push` block in the config.  That being said, you are correct about saturating the build queue and slowing down overall merge times.  It looks like we can use the `cron` syntax to schedule builds, so I'll look at doing this on a regular cadence.

File: .github/workflows/run-orch-smoketest.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bistline) [reply .github/workflows/run-orch-smoketest.yml]: [1cd34f7](https://github.com/broadinstitute/single_cell_portal_core/pull/1826/commits/1cd34f75a2dc2bd97db26643adcd7e5c7e8a50c7), [4f451c1](https://github.com/broadinstitute/single_cell_portal_core/pull/1826/commits/4f451c1612584ac3fa5489b4d0495b5197591cd7), [dcb212d](https://github.com/broadinstitute/single_cell_portal_core/pull/1826/commits/dcb212d91f6bdae04263f6c1d8334598f144c5ce), and [e74c1b6](https://github.com/broadinstitute/single_cell_portal_core/pull/1826/commits/e74c1b68e95e1f89ae3fa57fcbdb89024f047fb5)

**Review by ehanna4 — APPROVED** [review]
Functional review - worked as advertized

**Pull request merged**